File: RemoveDuplicates/RemoveDuplicates.java
/*
 * Problem:Remove Duplicates from Sorted Array
 * Given a sorted array, remove the duplicates 
 * in place such that each element appear only 
 * once and return the new length. Do not 
 * allocate extra space for another array, you
 * must do this in place with constant memory.
 * For example, Given input array A = [1,1,2],
 * Your function should return length = 2, and
 * A is now [1,2].
 * 
 * author Hyace
 * date:May 24, 2014
 *
 */
public class RemoveDuplicates{
	public static void main(String[] args){
		
	}
	public static int removeDuplicates(int[] arr){
		int n = arr.length;
		int point=0;
		for(int i=1;i<n;i++){
			if(arr[point]!=arr[i])
				arr[++point]=arr[i];
		}
		return point+1;
	}
}

File: RemoveDuplicates/removeDuplicates.py
__author__ = 'Hyace'
def removeDuplicates(arr):
    n=len(arr)
    index=0
    for i in range(1,n):
        if arr[index]!=arr[i]:
            index+=1
            arr[index]=arr[i]
    return index+1

arr=[1,1,1,2,2,3,3,4,4,4,5,5,6,6]
print removeDuplicates(arr)